File: 20-valid-parentheses/valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    // put into a key value pair for objects;
    // maybe a double loop?
    // check to see if first and last are open and closed
    const stack = [];

    for (let i = 0; i < s.length; i++) {
        if (s[i] === '{') {
            stack.push('}')
        }
        else if (s[i] === '[') {
            stack.push(']')
        }
        else if (s[i] === '(') {
            stack.push(')')
        }
        else if (stack.pop() !== s[i]) {
            return false
        }
    }
    return !stack.length
};

File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        obj = {
            '(': ')',
            '{': '}',
            '[': ']'
        }
        for bracket in s:
            if bracket in obj:
                stack.append(bracket)
            else:
                if len(stack) > 0:
                    temp = stack.pop()
                    if obj[temp] != bracket:
                            return False
                else:
                    return False
        if len(stack) == 0:
            return True
        else:
            return False